evalml/data_checks/outliers_data_check.py

```diff
@@ -50,20 +52,136 @@ def validate(self, X, y=None):
+        sample size. For each sample size, the shape and scale parameters of the
+        gamma distriubtion were estimated using maximum likelihood methods. The
+        set of estimate shape and scale parameters for different sample size were
+        then used to fit equations that relate these two parameters to the sample
+        size. These equations use a transendental logrithmic functional form that
+        provides a seventh order Taylor series approximation to the two true
+        functional relationships, and was estimated using least squares
+        regression.
+
+        Original credit goes to Jad Raad and Dan Putler of Alteryx.
+
+
+        Arguments:
+            num_records (int): The integer number of non-missing values in a column
+            pct_outliers (float): The percentage of potential outliers in a column
+        Returns:
+            float: The probability that no outliers are present in the column
+        """
+
+        # calculate the shape and scale parameters of the approximate
+        # gamma distribution given the number of records in the data.
+        # For both measures, the values are are from a least squares regression
+        # model
+        log_n = np.log(num_records)
+        log_shape = (
+            25.8218734380722 +
+            -29.2320460088643 * log_n +
+            14.8228030299864 * log_n ** 2 +
+            -4.08052512660036 * log_n ** 3 +
+            0.641429075842177 * log_n ** 4 +
+            -0.0571252717322226 * log_n ** 5 +
+            0.00268694343911156 * log_n ** 6 +
+            -5.19415149920567e-05 * log_n ** 7
+        )
+        shape_param = np.exp(log_shape)
+        log_scale = (
+            -19.8196822259052 +
+            8.5359212447622 * log_n +
+            -8.80487628113388 * log_n ** 2 +
+            2.27711870991327 * log_n ** 3 +
+            -0.344443407676357 * log_n ** 4 +
+            0.029820831994345 * log_n ** 5 +
+            -0.00136611527293756 * log_n ** 6 +
+            2.56727158170901e-05 * log_n ** 7
+        )
+        scale_param = np.exp(log_scale)
+
+        # calculate and return the probability of no true outliers for a gamma
+        # cumulative density function
+        prob_val = 1.0 - gamma.cdf(pct_outliers, shape_param, scale=scale_param)
+        return prob_val
+
+    @staticmethod
+    def _outlier_score(column, convert_column=False):
+        """Return a dictionary of high and low values of potential numeric outliers using the IQR method.
+
+        Original credit goes to Jad Raad and Dan Putler of Alteryx.
+
+        Arguments:
+            column (pd.Series): A column of data to check for outliers in
+            convert_column (bool): If True, convert column to np.int64, if possible.
+
+        Returns:
+            dict: Dictionary containing outlier information
+        """
+        column_nonan = column[~pd.isna(column)]
+
+        if column_nonan.shape[0] == 0:
+            return None
```

Comment:
Is this still necessary?  This seems like an implicit redundancy of highly null data check.  If not, it would be nice to snipe it and dedent the rest.

Reply:
I understand what you mean by this and the following comment, but I think this check is still nice to have here. In this method, we're checking if the length of the series is 0 because if it is, all of the other math that continues will error out (ex: np.percentile call) so we want to exit early. It is true that we have the highly null data check that will warn users if their data is all null, but our outlier data check still shouldn't error out if the user passes in data with all nan!